src/main/java/net/openhft/chronicle/network/ChronicleSocketChannelBuilder.java

```diff
@@ -0,0 +1,131 @@
+package net.openhft.chronicle.network;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.network.tcp.ChronicleSocket;
+import net.openhft.chronicle.network.tcp.ChronicleSocketChannel;
+import net.openhft.chronicle.network.tcp.ChronicleSocketChannelFactory;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+
+import static net.openhft.chronicle.core.io.Closeable.closeQuietly;
+import static net.openhft.chronicle.network.connection.TcpChannelHub.TCP_BUFFER;
+
+public class ChronicleSocketChannelBuilder {
+
+    @NotNull
+    private final InetSocketAddress socketAddress;
+    private int tcpBufferSize = Jvm.getInteger("tcp.client.buffer.size", TCP_BUFFER);
+    private int socketConnectionTimeoutMs = Jvm.getInteger("client.timeout", 500);
+    @Nullable
+    private InetSocketAddress localBinding;
+    private boolean tcpNoDelay = !TcpEventHandler.DISABLE_TCP_NODELAY;
+
+    public ChronicleSocketChannelBuilder(@NotNull InetSocketAddress socketAddress) {
+        this.socketAddress = socketAddress;
+    }
+
+    /**
+     * Set the TCP buffer size
+     *
+     * @param tcpBufferSize The TCP buffer size in bytes
+     * @return this
+     */
+    public ChronicleSocketChannelBuilder tcpBufferSize(int tcpBufferSize) {
+        this.tcpBufferSize = tcpBufferSize;
+        return this;
+    }
+
+    /**
+     * Set the socket connection timeout
+     *
+     * @param socketConnectionTimeoutMs the socket connection timeout in milliseconds
+     * @return this
+     */
+    public ChronicleSocketChannelBuilder socketConnectionTimeoutMs(int socketConnectionTimeoutMs) {
+        this.socketConnectionTimeoutMs = socketConnectionTimeoutMs;
+        return this;
+    }
+
+    /**
+     * Set whether to enable/disable <a href="https://en.wikipedia.org/wiki/Nagle%27s_algorithm">Nagle's algorithm</a> for this socket
+     *
+     * @param tcpNoDelay
+     * @return
+     */
+    public ChronicleSocketChannelBuilder tcpNoDelay(boolean tcpNoDelay) {
+        this.tcpNoDelay = tcpNoDelay;
+        return this;
+    }
+
+    /**
+     * Set the local socket to bind to
+     *
+     * @param localBinding The local socket to bind to, or null to not bind to any local socket
+     * @return this
+     */
+    public ChronicleSocketChannelBuilder localBinding(InetSocketAddress localBinding) {
+        this.localBinding = localBinding;
+        return this;
+    }
+
+    /**
+     * Open a socket channel with the builder's current settings
+     *
+     * @return The opened channel, or null if no channel could be opened
+     * @throws IOException if something goes wrong opening the channel
+     */
+    @Nullable
+    public ChronicleSocketChannel open() throws IOException {
```

Comment:
builder has an "open" method rather than a "build" method because it actually attempts to open the channel and returns null if it was not successful